crates/driver/src/tests/setup/blockchain.rs

```diff
@@ -30,6 +30,9 @@ pub struct Blockchain {
     pub trader_address: ethcontract::H160,
     pub trader_secret_key: SecretKey,
 
+    pub surplus_capturing_jit_order_owner_address: Option<ethcontract::H160>,
```

Comment:
> The public address can be recovered from the SecretKey so this is redundant. 
umm, so in theroy the L30 (`trader_address`) is also redundant right? :thinking: 

> One of them for signing orders that should get surplus and one for order without.
what is the benefit of it? but just having the address you want to sign the JIT orders should be enough for the test, then we can have 2 tests, one for the address to be whitelisted and another test with the address no whitelisted (as I have right now).

Reply:
> umm, so in theroy the L30 (trader_address) is also redundant right? 🤔

Yep.

> what is the benefit of it?

These test setups already have quite a bit of state. Reducing variable state to a minimum seems like a win to me. Currently there are a secret key, a public address and flags controlling whether an order is supposed to be surplus capturing or not.
Error cases could be: public jit order owner address that doesn't match the PK the order was signed with, conceptually 2 different JIT orders could be signed by the same key but one of them is supposed to get surplus but not the other due to the flags. Could be that these error cases currently cannot happen because every test only allows for a single JIT order but IMO a more general purpose testing framework should be able to handle multiple JIT orders where I would see the current code being error prone.
With the proposed change we'll always just have 2 private JIT owner keys (nothing optional), public addresses can be recovered from the key and the bool flag gets replaced by signing the order with one key or the other. Technically there is an argument to be made that this does not support different JIT order owners that should all capture or not capture surplus. But IMO the proposed change would lend itself more easily to being extended towards that than the current implementation.